subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -553,23 +576,53 @@ void end_report(void)
 K_APPMEM_PARTITION_DEFINE(ztest_mem_partition);
 #endif
 
+int ztest_list_tests(void)
+{
+	struct ztest_suite_node *ptr;
+	struct ztest_unit_test *test = NULL;
+	int test_count = 0;
+
+	TC_PRINT("Listing tests (suite::test)\n");
+	for (ptr = _ztest_suite_node_list_start; ptr < _ztest_suite_node_list_end; ++ptr) {
+		while ((test = ztest_get_next_test(ptr->name, test)) != NULL) {
+			TC_PRINT("    %s::%s\n", test->test_suite_name, test->name);
+			test_count++;
+		}
+	}
+
+	TC_PRINT("Number of tests: %d\n", test_count);
+
+	/* List tests only once */
+	list_tests = false;
+	return test_count;
+}
+
+bool ztest_should_suite_run(const void *state, struct ztest_suite_node *suite)
+{
+	bool run_suite = true;
+
+	if (test_args != NULL && strstr(test_args, suite->name) == NULL) {
```

Comment:
This could fail if you have `suite->name` as "test_suite" and `test_args` as "my_test_suite::*". I think you need additional logic to make sure that the suite is the full suite name.

Reply:
Done